daemon/src/projection.rs

```diff
@@ -971,7 +974,83 @@ impl db::ClosedCfdAggregate for Cfd {
             expiry_timestamp: Some(expiry_timestamp),
             counterparty: counterparty_network_identity,
             pending_settlement_proposal_price: None,
-            aggregated: empty_aggregated,
+            aggregated,
+            network,
+        }
+    }
+}
+
+impl db::FailedCfdAggregate for Cfd {
```

Comment:
❓Can we potentially create a generic aggregate for closed and failed?

Reply:
I don't see it at the moment because we have completely different expectations for closed and failed CFDs. The data that is stored is different, particularly in terms of the transactions that are attached to closed CFDs (always a lock transaction and a choice of settlement transactions) and that are not attached to failed CFDs. 

We could define a model with optional fields, but I'm not sure I would recommend that added complexity.